### PR TITLE
Fix torch.where(bool, int64, 0.5)

### PR DESCRIPTION
**Pull request by JackCaoG**
This is to fix https://github.com/pytorch/xla/issues/3105.

Case this failed is 
```
bool_t = torch.tensor([True, False], device=device)
true_t = torch.tensor([10, 20], device=device)

t1 = torch.where(bool_t, true_t, 0.5)
```

### CODE REVIEW

**Review by wonjoolee95 — APPROVED** [review]
LGTM 👍

**Comment by JackCaoG** [comment]
Need to resolve the conflict

**Review by miladm — CHANGES_REQUESTED** [review]
Thanks @JackCaoG!
LTGM

nit: shall we add this specific condition as a new test scenario to `TestWhere`?

File: torch_xla/csrc/tensor_methods.cpp [hunk 1]
```diff
@@ -2916,9 +2916,19 @@ void XLATensor::zero_(XLATensor& input) {
...
-      condition.GetIrValue(), input.GetIrValue(), other.GetIrValue()));
+                           const XLATensor& other,
+                           c10::optional<at::ScalarType> logical_element_type) {
+  if (logical_element_type.has_value()) {
+    return input.CreateFrom(
+        ir::ops::Where(
+            condition.GetIrValue(), input.GetIrValue(), other.GetIrValue(),
+            MakeXlaPrimitiveType(*logical_element_type, &input.GetDevice())),
+        logical_element_type);
+  } else {
```
Comment (miladm) [inline torch_xla/csrc/tensor_methods.cpp:2927]: nit: you can drop this else block given the previous return.

**Pull request closed without merging**